Mathlib/CategoryTheory/Generator/Coproduct.lean

```diff
@@ -0,0 +1,48 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import Mathlib.CategoryTheory.Generator.Basic
+import Mathlib.CategoryTheory.Limits.Shapes.Products
+
+/-!
+# The coproduct of a separating family of objects is separating
+
+If a family of objects `S : ι → C` in a category with zero morphisms
+is separating, then the coproduct of `S` is a separator in `C`.
+
+-/
+
+universe w v u
+
+namespace CategoryTheory
+
+open Limits
+
+variable {C : Type u} [Category.{v} C] [HasZeroMorphisms C]
+
+namespace IsSeparating
+
+open Classical in
+lemma isSeparator_of_isColimit_cofan
+    {ι : Type w} {S : ι → C} (hS : IsSeparating (Set.range S))
```

Comment:
Since `{ι : Type w} {S : ι → C}` are used in both of the lemmas in this file, perhaps you want to mark them as `variable` in line 24 as well?